src/site/content/en/blog/wordico/index.md

```diff
@@ -0,0 +1,426 @@
+<figcaption>The chat panel in Flash was pretty but complex.</figcaption>
+</figure>
+
+The HTML version, by comparison, is just a `<div>` with a fixed height and the overflow property set to hidden. Scrolling costs us nothing.
+
+<figure>
+{% Img src="image/T4FyVKpzu4WKF1kBNvXepbi08t52/tj47VhDgbeFG6PDiutB1.jpg", alt="The CSS box model at work.", width="284", height="323" %}
+<figcaption>The CSS box model at work.</figcaption>
+</figure>
+
+In cases like this - ordinary layout tasks - HTML and CSS outshine Flash.
+
+## Can you hear me now?
+
+We struggled with the `<audio>` tag - it simply wasn't capable of playing short sounds effects repeatedly in certain browsers. We tried two workarounds. First, we padded the sound files with dead air to make them longer. Then we tried alternating playback across multiple audio channels. Neither technique was completely effective or elegant.
+
+Ultimately we decided to roll our own Flash audio player and use HTML5 audio as a fallback. Here's the basic code in Flash:
+
+```js
+var sounds = new Array();
+
+function playSound(path:String):void {
+var sound:Sound = sounds[path];
+
+if (sound == null) {
+sound = new Sound();
+sound.addEventListener(Event.COMPLETE, function (evt:Event) {
+    sound.play();
+});
+sound.load(new URLRequest(path));
+sounds[path] = sound;
+}
+else {
+sound.play();
+}
+}
+
+ExternalInterface.addCallback("playSound", playSound);
+```
+
+In JavaScript, we attempt to detect the embedded Flash player.  If that fails, we create an `<audio>` node for each sound file:
+
+```js
+function play(String soundId) {
+var src = "/audio/" + soundId + ".mp3";
+
+// Flash
+try {
+var swf = window["swfplayer"] || document["swfplayer"];
+swf.playSound(src);
+}
+// or HTML5 audio
+catch (e) {
+var sound = document.getElementById(soundId);
+if (sound == null || sound == undefined) {
+    var sound = document.createElement("audio");
+    sound.id = soundId;
+    sound.src = src;
+    document.body.appendChild(sound);
+}
+sound.play();
+}
+}
+```
+
+Note that this works for MP3 files only - we never bothered to support OGG. We hope the industry will settle on a single format in the near future.
+
+## Poll position
+
+We use the same technique in HTML5 as we did in Flash to refresh the game state: every 10 seconds, the client asks the server for updates. If the game state has changed since the last poll, the client receives and handles the changes; otherwise, nothing happens. This traditional polling technique is acceptable, if not quite elegant. However, we'd like to switch to <a href="">[long polling](http://en.wikipedia.org/wiki/Long_polling#Long_polling)</a> or <a href="">[WebSockets](http://en.wikipedia.org/wiki/Web_Sockets)</a> as the game matures and users come to expect real-time interaction over the network. WebSockets, in particular, would present many opportunities to enhance the game play.
```

Comment:
```suggestion
We use the same technique in HTML5 as we did in Flash to refresh the game state: every 10 seconds, the client asks the server for updates. If the game state has changed since the last poll, the client receives and handles the changes; otherwise, nothing happens. This traditional polling technique is acceptable, if not quite elegant. However, we'd like to switch to [long polling](http://en.wikipedia.org/wiki/Long_polling#Long_polling) or [WebSockets](http://en.wikipedia.org/wiki/Web_Sockets) as the game matures and users come to expect real-time interaction over the network. WebSockets, in particular, would present many opportunities to enhance the game play.
```